pkg/features/kube_features.go

```diff
@@ -296,45 +294,45 @@ func init() {
 // To add a new feature, define a key for it above and add it here. The features will be
 // available throughout Kubernetes binaries.
 var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureSpec{
-	AppArmor:                                    {Default: true, PreRelease: utilfeature.Beta},
-	DynamicKubeletConfig:                        {Default: true, PreRelease: utilfeature.Beta},
-	ExperimentalHostUserNamespaceDefaultingGate: {Default: false, PreRelease: utilfeature.Beta},
```

Comment:
for reference, this change was made to "Beta" in error in this PR:
https://github.com/kubernetes/kubernetes/pull/40216/files#diff-84e72664a7cc9add01954d3a7b8cbb45R75

the prior behavior was alpha, and it looks to have gone to beta with off by default by accident in the above PR.

there is no way the feature could have worked properly.

that said, we should add a release note to this PR that says the feature gate for `ExperimentalHostUserNamespaceDefaulting` has been removed.

Reply:
I'm not sure about the deprecation policy for alpha feature gates, but dropping it directly may cause components to crash since the field is no longer recognized (even for `ExperimentalHostUserNamespaceDefaulting=false`). Is there a way to add a deprecation warning before removing it?

Reply:
@yujuhong adding the deprecation makes sense to me.

@vikaschoudhary16 can you mark the feature gate deprecated similar to `ReadOnlyAPIDataVolumes`?  the release note can note that the feature gate is deprecated, not supported, and will be removed next release.